website/projects/spoke/src/app/core/state/google-analytics.service.ts

```diff
@@ -0,0 +1,227 @@
+import { Injectable, Inject, InjectionToken, isDevMode } from '@angular/core';
+import { NGX_GOOGLE_ANALYTICS_SETTINGS_TOKEN } from 'ngx-google-analytics';
+import { IGoogleAnalyticsSettings } from 'ngx-google-analytics';
+import { GaActionEnum } from 'ngx-google-analytics';
+import { DOCUMENT } from '@angular/common';
+import { NGX_GTAG_FN } from 'ngx-google-analytics';
+import { GtagFn } from 'ngx-google-analytics';
+
+    
+function getGtagFnNull(): GtagFn {
+  return {} as GtagFn;
+};
+    
+const NGX_GTAG_FN_NULL = new InjectionToken<GtagFn>('ngx-gtag-fn', {
+  factory: () => getGtagFnNull()
+});
+    
+const token = localStorage['ALLOW_TELEMETRY'] !== 'false' ? NGX_GTAG_FN : NGX_GTAG_FN_NULL
```

Comment:
This can/will change at runtime. You will need to test this _every_ time the gtag function is called. Update your custom gtagFunction to do this test at function call time.